client/src/components/settings/certification.tsx

```diff
@@ -13,60 +12,41 @@ import ProjectPreviewModal from '../../templates/Challenges/components/project-p
 import { openModal } from '../../templates/Challenges/redux/actions';
 import {
   projectMap,
-  legacyProjectMap
+  legacyProjectMap,
+  fullProjectMap,
+  ProjectMap,
+  LegacyProjectMap
 } from '../../resources/cert-and-project-map';
 import { FlashMessages } from '../Flash/redux/flash-messages';
 import ProjectModal from '../SolutionViewer/project-modal';
 import { FullWidthRow, Spacer } from '../helpers';
 import { SolutionDisplayWidget } from '../solution-display-widget';
-import SectionHeader from './section-header';
+import { certSlugTypeMap } from '../../../../config/certification-settings';
 
 import './certification.css';
+import {
+  ClaimedCertifications,
+  CompletedChallenge,
+  User
+} from '../../redux/prop-types';
+import { createFlashMessage } from '../Flash/redux';
+import { verifyCert } from '../../redux/settings/actions';
+import SectionHeader from './section-header';
 
 configureAnchors({ offset: -40, scrollDuration: 0 });
 
-const propTypes = {
-  completedChallenges: PropTypes.arrayOf(
-    PropTypes.shape({
-      id: PropTypes.string,
-      solution: PropTypes.string,
-      githubLink: PropTypes.string,
-      challengeType: PropTypes.number,
-      completedDate: PropTypes.number,
-      challengeFiles: PropTypes.array
-    })
-  ),
-  createFlashMessage: PropTypes.func.isRequired,
-  is2018DataVisCert: PropTypes.bool,
-  isApisMicroservicesCert: PropTypes.bool,
-  isBackEndCert: PropTypes.bool,
-  isDataAnalysisPyCertV7: PropTypes.bool,
-  isDataVisCert: PropTypes.bool,
-  isCollegeAlgebraPyCertV8: PropTypes.bool,
-  isFrontEndCert: PropTypes.bool,
-  isFrontEndLibsCert: PropTypes.bool,
-  isFullStackCert: PropTypes.bool,
-  isHonest: PropTypes.bool,
-  isInfosecCertV7: PropTypes.bool,
-  isInfosecQaCert: PropTypes.bool,
-  isJsAlgoDataStructCert: PropTypes.bool,
-  isMachineLearningPyCertV7: PropTypes.bool,
-  isQaCertV7: PropTypes.bool,
-  isRelationalDatabaseCertV8: PropTypes.bool,
-  isRespWebDesignCert: PropTypes.bool,
-  isSciCompPyCertV7: PropTypes.bool,
-  openModal: PropTypes.func,
-  t: PropTypes.func.isRequired,
-  username: PropTypes.string,
-  verifyCert: PropTypes.func.isRequired
-};
-
 const mapDispatchToProps = {
   openModal
 };
 
-const certifications = Object.keys(projectMap);
-const legacyCertifications = Object.keys(legacyProjectMap);
+// Safety: projectMap definitely has projectMap keys,
+// and we are only interested in these keys
+const certifications = Object.keys(projectMap) as Array<keyof ProjectMap>;
+// Safety: legacyProjectMap definitely has legacyProjectMap keys,
+// and we are only interested in these keys
+const legacyCertifications = Object.keys(legacyProjectMap) as Array<
+  keyof LegacyProjectMap
+>;
```

Comment:
What do you think of creating a helper function, that creates the types for the keys?

This way if someone need to type keys of an Object, and don't know the types, they can use the utility function and get the types of the object. I checked the codebase and there are some cases which create `Object.keys() as typeof keys` too, and I am slightly worried that a future dev will struggle with these types 🤔

```suggestion
const objectKeys = <Obj extends Record<string, unknown>, Keys extends (keyof Obj)[]>(obj: Obj): Keys => {
  return Object.keys(obj) as Keys;
}

const certifications = objectKeys(projectMap);
const legacyCertifications = objectKeys(legacyProjectMap);
```

Reply:
I would not like a utility function that asserts a generic's type.

Anywhere we have an assertion, we should be justifying it.

Reply:
Valid point, and I don't think the utility function will stop a bug, all it would do is show keys values when you hover over the objectkeys, which can be done through checking definiton.

![image](https://user-images.githubusercontent.com/88248797/228291281-4df74e06-58dc-458c-9ba2-c2259107f4c8.png)